packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -262,8 +262,8 @@ class UpgradeCommandRunner {
   ///
   /// If the user is on a deprecated channel, attempts to migrate them off of
   /// it.
-  Future<void> upgradeChannel(FlutterVersion flutterVersion) async {
-    globals.printStatus('Upgrading Flutter from $workingDirectory...');
+  Future<void> upgradeChannel(FlutterVersion flutterVersion, String upstreamRevision) async {
+    globals.printStatus('Upgrading Flutter to $upstreamRevision from ${flutterVersion.frameworkRevision} in $workingDirectory...');
```

Comment:
I just realized that using `flutterVersion.frameworkRevision` is probably not what the PR is trying to do:

Take a look at these lines of code:
https://github.com/flutter/flutter/blob/13896b3bd1858687a06b1639d60c877206170df8/packages/flutter_tools/lib/src/version.dart#L170-L183

My understanding is that `flutterVersion.frameworkVersion` should be used, but is there anyway the corresponding version of the `upstreamRevision` can be obtained?

Reply:
Thats a good point, it is probably printing out the commit hash right now. We should know what the tag we're checking out is

Reply:
This might be something of interest in that case:
https://github.com/flutter/flutter/blob/13896b3bd1858687a06b1639d60c877206170df8/packages/flutter_tools/lib/src/version.dart#L77

Reply:
> Thats a good point, it is probably printing out the commit hash right now. We should know what the tag we're checking out is

@jonahwilliams You're right. #68866 was merged yesterday which uses `flutterVersion.frameworkRevision`:
```
$ flutter upgrade --verify-only
A new version of Flutter is available on channel master

The latest revision: 18341efa2fb8c3570ca32e71d1a9da74e8b2cbf5
Your current version: 8b9e9680d24ffd4b3dbfa216f5f64a0490088f45

To upgrade now, run "flutter upgrade".
```

Reply:
@RoyARG02 @jonahwilliams Thanks for reviewing it for you.
I understand that using flutterVersion.frameworkRevision is a mistake.
Thank you.

I will modify it to use frameworkVersion.

Reply:
I have modified it to use frameworkVersion.
I also rewrote the test and it was successful!
Please give you a review.

Reply:
I'm taking a look now.